### PR TITLE
Duplicated startxref and failed to search for AcroForm.

### PR DESCRIPTION
**Pull request by therpobinski**
Fix for diplicate `startxref` and fix in finding last ID of `AcroForm`.

### CODE REVIEW

**Comment by therpobinski** [comment]
Hello, I see that the changes have not been mixed yet. As I mentioned in the problem, there are some errors in testing. The truth is that those errors happened when I did a `pull` of the last changes of its default branch. Could you tell me what I can do to correct them?

File: package.json [hunk 1]
```diff
@@ -63,5 +63,8 @@
         "jest": "^24.5.0",
         "node-forge": "^0.7.6",
         "pdfkit": "^0.10.0"
+    },
+    "dependencies": {
+        "pdf-lib": "^1.2.1"
```
Comment (vbuch) [inline package.json:68]: The library does not have explicit dependencies. We use peer dependencieso that choosing your libs is up to you. We only bundle our code and rely on the user to provide the needed deps.

File: src/signpdf.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ export class SignPdf {
             bagType: forge.pki.oids.pkcs8ShroudedKeyBag,
         })[forge.pki.oids.pkcs8ShroudedKeyBag];
 
-        const privateKey = keyBags[0].key;
+        const privateKey = (keyBags[1] || keyBags[0]).key;
```
Comment (vbuch) [inline src/signpdf.js:102]: Not quite convinced by this line. Actually matching should be done by key id here and not just guessing it's the first one.

File: src/signpdf.test.js [hunk 3]
```diff
@@ -163,18 +167,58 @@ describe('Test signing', () => {
...
         const secondP12Buffer = fs.readFileSync(`${__dirname}/../resources/withpass.p12`);
         let signedPdfBuffer = fs.readFileSync(`${__dirname}/../resources/signed-once.pdf`);
-        signedPdfBuffer = plainAddPlaceholder({
-            pdfBuffer: signedPdfBuffer,
-            reason: 'second',
-            signatureLength: 1592,
+        signedPdfBuffer = plainAddPlaceholder({ 
+          pdfBuffer: signedPdfBuffer, 
+          reason: 'second',
+          signatureLength: secondP12Buffer.length
```
Comment (vbuch) [inline src/signpdf.test.js:183]: The length of the p12 buffer does not relate to the length of the signature. Read the note here: https://github.com/vbuch/node-signpdf/tree/6d74d98d2038837d29055f346be7dd6b758158ac#append-a-signature-placeholder

**Comment by therpobinski** [comment]
I thought you never replied to this `PR`, I never got a notification. By reviewing your comments I will continue to resolve them.
If I have a question with this [comment](https://github.com/vbuch/node-signpdf/pull/77#discussion_r459234848), do you have any ideas or sources on how to get the length of the `signature`?

**Comment by vbuch** [comment]
@therpobinski read the note I linked. Just test it, Given the same certificate, the signature's lenght won't change. So just test it and find the needed length.

**Pull request closed without merging**